test/e2e/resourceinterpreter_test.go

```diff
@@ -422,4 +422,191 @@ end`,
 			}
 		})
 	})
+
+	ginkgo.Context("InterpreterOperation Retain testing", func() {
+		var waitTime time.Duration
+		var updatedPaused bool
+
+		ginkgo.BeforeEach(func() {
+			waitTime = 5 * time.Second
+			updatedPaused = true
+			customization = testhelper.NewResourceInterpreterCustomization(
+				"interpreter-customization"+rand.String(RandomStrLength),
+				configv1alpha1.CustomizationTarget{
+					APIVersion: "apps/v1",
+					Kind:       "Deployment",
+				},
+				configv1alpha1.CustomizationRules{
+					Retention: &configv1alpha1.LocalValueRetention{
+						LuaScript: `
+function Retain(desiredObj, observedObj)
+  desiredObj.spec.paused = observedObj.spec.paused
+  return desiredObj
+end`,
+					},
+				})
+		})
+
+		ginkgo.It("Retain testing", func() {
+			ginkgo.By("update deployment's spec.paused to true", func() {
+				clusterClient := framework.GetClusterClient(targetCluster)
+				gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
+				var memberDeploy *appsv1.Deployment
+				framework.WaitDeploymentPresentOnClusterFitWith(targetCluster, deployment.Namespace, deployment.Name,
+					func(deployment *appsv1.Deployment) bool {
+						memberDeploy = deployment
+						return true
+					})
+				memberDeploy.Spec.Paused = updatedPaused
+				framework.UpdateDeployment(clusterClient, memberDeploy)
+			})
+
+			// Wait executeController to reconcile then check if it is retained
+			time.Sleep(waitTime)
```

Comment:
I don't think `Sleep` is a good way. Could we design the test as follower:
1. Create retention like
```lua
desiredObj.FooField = observedObj.FooField
```

2. Then update object in controller panel from
```yaml
FooField: foo
BarFiled: bar
```
to 
```yaml
FooField: foo1
BarFiled: bar1
```

3. Wait object in member cluster to be:
`````yaml
FooField: foo
BarFiled: bar1
```

That says our update is going into effect, and our interpreter works.

Reply:
It looks great. Because this modification also involves other cases:

https://github.com/karmada-io/karmada/blob/master/test/e2e/resourceinterpreter_test.go#L106-L118

So I will send a pr later,